File: src/lib/rubeno/lib/base_act.rb
module Rubeno
  # BaseAct extends BaseAction to support AAA pattern.
  class BaseAct < BaseAction
    def initialize(name, act_cb)
      super(name, act_cb)
    end
    
    # Alias perform_action to perform_act for AAA pattern
    def perform_act(store, act_cb, test_resource)
      perform_action(store, act_cb, test_resource)
    end
    
    # Alias test to act for AAA pattern
    def act(store, test_resource_configuration)
      test(store, test_resource_configuration)
    end
  end
end

File: src/lib/tiposkripto/src/BaseAct.ts
import { BaseAction } from "./BaseAction.js";
import { TestTypeParams_any } from "./CoreTypes.js";

/**
 * BaseAct extends BaseAction to support AAA pattern.
 * It reuses all Action functionality but with AAA naming.
 */
export class BaseAct<I extends TestTypeParams_any> extends BaseAction<I> {
  constructor(name: string, actCB: (xyz: I["iselection"]) => I["then"]) {
    super(name, actCB);
  }

  // Alias performAction to performAct for AAA pattern
  async performAct(
    store: I["istore"],
    actCB: (x: I["iselection"]) => I["then"],
    testResource
  ) {
    return super.performAction(store, actCB, testResource);
  }

  // Alias test to act for AAA pattern
  async act(
    store: I["istore"],
    testResourceConfiguration
  ) {
    return super.test(store, testResourceConfiguration);
  }
}

export type IActs<I extends TestTypeParams_any> = Record<string, BaseAct<I>>;
